src/kit.ts

```diff
@@ -1370,7 +1370,8 @@ export function kitChangeNeedsClean(newKit: Kit, oldKit: Kit | null): boolean {
         vs: k.visualStudio,
         vsArch: k.visualStudioArchitecture,
         tc: k.toolchainFile,
-        preferredGenerator: k.preferredGenerator ? k.preferredGenerator.name : null
+        preferredGeneratorName: k.preferredGenerator ? k.preferredGenerator.name : null,
+        preferredGeneratorPlatform: k.preferredGenerator && k.preferredGenerator.platform ? k.preferredGenerator.platform : null
```

Comment:
Since we're already checking generator name and platform, should we check generator toolset as well? Or is that not important to check for cleaning?

Reply:
@jophippe @snehara99 I don't think it's crucial, but if it's simple to add I think that it could be worthwhile.

Reply:
updated